p2p/security/tls/transport_test.go

```diff
@@ -175,6 +176,111 @@ func TestHandshakeSucceeds(t *testing.T) {
 	})
 }
 
+func TestHandshakeWithNextProtoSucceeds(t *testing.T) {
+	var clientMuxerList = [][]protocol.ID{{}, {"muxer1/1.0.0", "muxer2/1.0.1"}, {"muxer1"}, {}, {"muxer1"}}
+	var serverMuxerList = [][]protocol.ID{{}, {"muxer2/1.0.1", "muxer1/1.0.0"}, {}, {"muxer1"}, {"muxer2"}}
+	var expectedMuxers = []string{"", "muxer2/1.0.1", "", "", ""}
+	numMuxers := len(clientMuxerList)
+	var expectedMuxer string
+
+	clientID, clientKey := createPeer(t)
+	serverID, serverKey := createPeer(t)
+
+	handshake := func(t *testing.T, clientTransport *Transport, serverTransport *Transport) {
+		clientInsecureConn, serverInsecureConn := connect(t)
+
+		serverConnChan := make(chan sec.SecureConn)
+		go func() {
+			serverConn, err := serverTransport.SecureInbound(context.Background(), serverInsecureConn, "")
+			require.NoError(t, err)
+			serverConnChan <- serverConn
+		}()
+
+		clientConn, err := clientTransport.SecureOutbound(context.Background(), clientInsecureConn, serverID)
+		require.NoError(t, err)
+		defer clientConn.Close()
+
+		var serverConn sec.SecureConn
+		select {
+		case serverConn = <-serverConnChan:
+		case <-time.After(250 * time.Millisecond):
+			t.Fatal("expected the server to accept a connection")
+		}
+		defer serverConn.Close()
+
+		require.Equal(t, clientConn.LocalPeer(), clientID)
+		require.Equal(t, serverConn.LocalPeer(), serverID)
+		require.True(t, clientConn.LocalPrivateKey().Equals(clientKey), "client private key mismatch")
+		require.True(t, serverConn.LocalPrivateKey().Equals(serverKey), "server private key mismatch")
+		require.Equal(t, clientConn.RemotePeer(), serverID)
+		require.Equal(t, serverConn.RemotePeer(), clientID)
+		require.True(t, clientConn.RemotePublicKey().Equals(serverKey.GetPublic()), "server public key mismatch")
+		require.True(t, serverConn.RemotePublicKey().Equals(clientKey.GetPublic()), "client public key mismatch")
+		require.Equal(t, clientConn.ConnState().NextProto, expectedMuxer)
+		// exchange some data
+		_, err = serverConn.Write([]byte("foobar"))
+		require.NoError(t, err)
+		b := make([]byte, 6)
+		_, err = clientConn.Read(b)
+		require.NoError(t, err)
+		require.Equal(t, string(b), "foobar")
+	}
+
+	// Iterate through the NextProto combinations.
+	for i := 0; i < numMuxers; i++ {
+		expectedMuxer = expectedMuxers[i]
+		clientTransport, err := New(clientKey, clientMuxerList[i])
+		require.NoError(t, err)
+		serverTransport, err := New(serverKey, serverMuxerList[i])
+		require.NoError(t, err)
+
+		// Use standard transports with default TLS configuration
+		t.Run("standard TLS with extension not critical", func(t *testing.T) {
+			handshake(t, clientTransport, serverTransport)
+		})
+
+		t.Run("standard TLS with extension critical", func(t *testing.T) {
+			extensionCritical = true
+			t.Cleanup(func() { extensionCritical = false })
+
+			handshake(t, clientTransport, serverTransport)
+		})
```

Comment:
I don't think this is worth doing here. Let's focus on testing the muxer selection in this test case, and just use whatever the default is in this test case.

Reply:
done